core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -471,8 +472,16 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
 
         for (thirdCiv in civInfo.getKnownCivs()) {
             // Our ally city states make peace with us
-            if (thirdCiv.getAllyCiv() == civInfo.civName && thirdCiv.isAtWarWith(otherCiv))
-                thirdCiv.getDiplomacyManager(otherCiv).makePeace()
+            if (thirdCiv.getAllyCiv() == civInfo.civName && thirdCiv.isAtWarWith(otherCiv)) {
+                val thirdCivDiplo = thirdCiv.getDiplomacyManager(otherCiv)
+                thirdCivDiplo.makePeace()
+                val tradeLogic = TradeLogic(thirdCiv, otherCiv)
+
+                tradeLogic.currentTrade.ourOffers.add(TradeOffer(Constants.peaceTreaty, TradeType.Treaty))
+                tradeLogic.currentTrade.theirOffers.add(TradeOffer(Constants.peaceTreaty, TradeType.Treaty))
+                thirdCivDiplo.trades.add(tradeLogic.currentTrade)
+                thirdCivDiplo.otherCivDiplomacy().trades.add(tradeLogic.currentTrade.reverse())
```

Comment:
Oh, interesting! So if I (A) make peace with my enemy (B) and I have an ally (C) who made war together with me, then C will send B a peace offer, and B can potentially say "no" to that! That'll make for an interesting case 🤔

Reply:
That is not possible, we are not offering the peace Trade here. We only put it in place. Neither civ has the chance to accept or reject.

Reply:
Gotcha